WireDomain/Sources/WireDomain/Repositories/Conversations/ConversationLocalStore.swift

```diff
@@ -0,0 +1,437 @@
+
+            commonUpdate(from: remoteConversation, for: $0, isFederationEnabled: isFederationEnabled)
+            updateConversationStatus(from: remoteConversation, for: $0)
+
+            isInitialFetch ?
+                assignMessageProtocol(from: remoteConversation, for: $0) :
+                updateMessageProtocol(from: remoteConversation, for: $0)
+
+            return ($0, $0.mlsGroupID)
+        }
+
+        await updateMLSStatus(from: remoteConversation, for: conversation)
+
+        await context.perform { [self] in
+            if isInitialFetch {
+                /// we just got a new conversation, we display new conversation header
+                conversation.appendNewConversationSystemMessage(
+                    at: .distantPast,
+                    users: conversation.localParticipants
+                )
+
+                /// Slow synced conversations should be considered read from the start
+                conversation.lastReadServerTimeStamp = conversation.lastModifiedDate
+            }
+
+            /// If we discover this group is actually a fake one on one,
+            /// then we should link the one on one user.
+            linkOneOnOneUserIfNeeded(for: conversation)
+        }
+    }
+
+    /// Updates or creates a conversation of type `1:1` locally.
+    ///
+    /// __What is a `1:1` conversation ?__
+    ///
+    /// The implementation differs between Proteus-based conversations and MLS-based conversations.
+    ///
+    /// *Proteus (not in a team)*
+    ///
+    /// When a connection is created it also creates an associated conversation. As the name implies a 1:1 conversation is always between two users which can't be changed.
+    ///
+    /// *Proteus (in a team)*
+    ///
+    /// A team 1:1 is a conversation between two users that belong to the same team.
+    ///
+    /// *MLS (whether in team or not in team)*
+    ///
+    /// MLS 1:1 conversations always implicitly exist when there’s a connection between two users, either via a connection or indirectly when the two users belong to the same team.
+    /// Therefore the conversation doesn’t need to be created but the underlying MLS group needs to be established.
+    ///
+    /// __What is `Federation` ?__
+    ///
+    /// Businesses that use Wire in their organization have 2 choices, either deploy their teams in the Wire cloud or install an on-prem backend.
+    /// On-prem backends are isolated islands that before could not communicate with users from other on-prem islands or Wire cloud users.
+    /// Federation allows islands to be connected. Communication can be established between users from different backends the same way as on a single backend.
+    ///
+    /// - Parameter remoteConversation: The conversation object received from backend.
+    /// - Parameter removeConversationID: The conversation ID received from backend.
+    /// - Parameter isFederationEnabled: A flag indicating whether a federation is enabled.
```

Comment:
we're repeating the definition of Federation, maybe we move this back to a docC within WireDomain and point to it instead

Reply:
done, I've added some documentation for conversations and federation using docc